src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1020,22 +1020,30 @@ private void MergeBlobAndDeleteVector(DocumentsOperationContext context, DbIdsHo
             }
         }
 
-        internal static long InternalGetCounterValue(BlittableJsonReaderObject.RawBlob localCounterValues, string docId, string counterName)
+        internal static long InternalGetCounterValue(BlittableJsonReaderObject.RawBlob localCounterValues, string docId, string counterName, bool capOnOverflow = false)
         {
             Debug.Assert(localCounterValues != null);
             var count = localCounterValues.Length / SizeOfCounterValues;
             long value = 0;
-
+            long uncheckedValue = 0;
             try
             {
                 for (int index = 0; index < count; index++)
                 {
+                    if (capOnOverflow)
+                        uncheckedValue = unchecked(value + ((CounterValues*)localCounterValues.Ptr)[index].Value);
```

Comment:
This is confusing, better to run the sum operation again on the `catch` clause and use that there